components/dashboard/src/data/configurations/delete-configuration-mutation.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useMutation } from "@tanstack/react-query";
+import { configurationClient } from "../../service/public-api";
+
+const BASE_KEY = "configurations";
+
+export const useDeleteConfiguration = (configurationId: string) => {
+    return useMutation<void, Error>(getConfigurationQueryKey(configurationId), async () => {
+        await configurationClient.deleteConfiguration({
+            configurationId,
+        });
```

Comment:
we need to make sure to invalidate the correct queries.

Reply:
Yah, I think we can invalidate the list querys as a whole, and then the individual config query.

```ts
// maybe wrap this in a fn we export from `configuration-queryies.ts` file to get the key
queryClient.invalidateQueries({ queryKey: ['configurations', 'list'] })
queryClient.invalidateQueries({ queryKey: ['configurations', { configurationId ] })
```